dynamicio/config/pydantic/config.py

```diff
@@ -0,0 +1,92 @@
+# pylint: disable=no-member, no-self-argument, unused-argument
+"""Pydantic schema for YAML files"""
+
+import typing
+
+import pydantic
+
+import dynamicio.config.pydantic.io_resources as env_spec
+import dynamicio.config.pydantic.table_schema as table_spec
+
+
+class IOBinding(pydantic.BaseModel):
+    """A binding for a single i/o object"""
+
+    name: str = pydantic.Field(alias="__binding_name__")
+    environments: typing.Mapping[str, env_spec.IOEnvironment]
+    dynamicio_schema: typing.Union[table_spec.DataframeSchema, table_spec.DataframeSchemaRef, None] = pydantic.Field(default=None, alias="schema")
```

Comment:
Can you import the typehint you want at the start to make types more readable? 

E.g.: 

```python 
from  typing import Mapping

...
environments: Mapping[str, env_spec.IOEnvironment]
...
```